rewrite-java/src/main/java/org/openrewrite/java/ChangePackage.java

```diff
@@ -349,4 +383,24 @@ private boolean isTargetRecursivePackageName(String packageName) {
         }
 
     }
+
+    @Value
+    @EqualsAndHashCode(callSuper = false)
+    private static class ReferenceChangePackageVisitor extends TreeVisitor<Tree, ExecutionContext> {
+        Set<Reference> matches;
+        Reference.Renamer renamer;
+        String newPackageName;
+
+        @Override
+        public @Nullable Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
```

Comment:
In `ReferenceChangeTypeVisitor` we override `preVisit()`. That (or `postVisit()`) is generally preferred when applicable.